redis/src/main/scala/zio/redis/options/Streams.scala

```diff
@@ -75,7 +75,7 @@ trait Streams {
 
   type NoAck = NoAck.type
 
-  case class MaxLen(approximate: Boolean, count: Long)
+  case class StreamMaxLen(approximate: Boolean, count: Long)
```

Comment:
Not necessary related to this PR, but I'd make these sealed.

Reply:
Done and made a lot of other case classes sealed. Can you maybe explain to me why the compiler refused to accept `final` case classes here that would even be better than just sealed or do I get something wrong?